app/public/websocket.js

```diff
@@ -13,7 +13,16 @@ const WS = (() => {
     }
     const onReceive = (func) => {
         webSocket.onmessage = (event) => {
-            func(JSON.parse(event.data));
+            const data=JSON.parse(event.data)
+            //Modified to handle the situation when game ends   --xmy
+            if (data.messageType === 'gameEnd') {
+                const result = data.result;
+                const gameResultElement = document.getElementById('game-result');
```

Comment:
No such element exists yet so this will throw an error. I love your initiative here, though.
Maybe comment out these lines, or just call some `onGameEnd()` function that we can hook up later